include/libopencm3/gd32/f1x0/rcc.h

```diff
@@ -0,0 +1,507 @@
+	RCC_TIM14	= _REG_BIT(0x1C, 8),
+	RCC_WWDG	= _REG_BIT(0x1C, 11),
+	RCC_SPI2	= _REG_BIT(0x1C, 14),
+	RCC_SPI3	= _REG_BIT(0x1C, 15),
+	RCC_USART2	= _REG_BIT(0x1C, 17),
+	RCC_I2C1	= _REG_BIT(0x1C, 21),
+	RCC_I2C2	= _REG_BIT(0x1C, 22),
+	RCC_USB		= _REG_BIT(0x1C, 23),
+	RCC_PWR		= _REG_BIT(0x1C, 28),
+	RCC_DAC		= _REG_BIT(0x1C, 29),
+	RCC_CEC		= _REG_BIT(0x1C, 30),
+};
+
+enum rcc_periph_rst {
+
+	/* Advanced peripherals */
+	RST_BACKUPDOMAIN = _REG_BIT(0x20, 16),/* BDCR[16] */
+
+	/* AHB peripherals */
+	RST_GPIOA	= _REG_BIT(0x28, 17),
+	RST_GPIOB	= _REG_BIT(0x28, 18),
+	RST_GPIOC	= _REG_BIT(0x28, 19),
+	RST_GPIOD	= _REG_BIT(0x28, 20),
+	RST_GPIOE	= _REG_BIT(0x28, 21),
+	RST_GPIOF	= _REG_BIT(0x28, 22),
+	RST_TSC		= _REG_BIT(0x28, 24),
+
+	/* APB2 peripherals */
+	RST_SYSCFG	= _REG_BIT(0x0C, 0),
+	RST_ADC		= _REG_BIT(0x0C, 9),
+	RST_ADC1	= _REG_BIT(0x0C, 9), /* Compatibility alias */
+	RST_TIM1	= _REG_BIT(0x0C, 11),
+	RST_SPI1	= _REG_BIT(0x0C, 12),
+	RST_USART1	= _REG_BIT(0x0C, 14),
+	RST_TIM15	= _REG_BIT(0x0C, 16),
+	RST_TIM16	= _REG_BIT(0x0C, 17),
+	RST_TIM17	= _REG_BIT(0x0C, 18),
+
+	/* APB1 peripherals */
+	RST_TIM2	= _REG_BIT(0x10, 0),
+	RST_TIM3	= _REG_BIT(0x10, 1),
+	RST_TIM6	= _REG_BIT(0x10, 4),
+	RST_TIM14	= _REG_BIT(0x10, 8),
+	RST_WWDG	= _REG_BIT(0x10, 11),
+	RST_SPI2	= _REG_BIT(0x10, 14),
+	RST_SPI3	= _REG_BIT(0x10, 15),
+	RST_USART2	= _REG_BIT(0x10, 17),
+	RST_I2C1	= _REG_BIT(0x10, 21),
+	RST_I2C2	= _REG_BIT(0x10, 22),
+	RST_USB		= _REG_BIT(0x10, 23),
+	RST_PWR		= _REG_BIT(0x10, 28),
+	RST_DAC		= _REG_BIT(0x10, 29),
+	RST_CEC		= _REG_BIT(0x10, 30),
+};
+
+#include <libopencm3/stm32/common/rcc_common_all.h>
+
+BEGIN_DECLS
+
+void rcc_osc_ready_int_clear(enum rcc_osc osc);
+void rcc_osc_ready_int_enable(enum rcc_osc osc);
+void rcc_osc_ready_int_disable(enum rcc_osc osc);
+int rcc_osc_ready_int_flag(enum rcc_osc osc);
+void rcc_css_int_clear(void);
+int rcc_css_int_flag(void);
+void rcc_osc_on(enum rcc_osc osc);
+void rcc_osc_off(enum rcc_osc osc);
+void rcc_css_enable(void);
+void rcc_css_disable(void);
+void rcc_set_sysclk_source(uint32_t clk);
+void rcc_set_pll_multiplication_factor(uint32_t mul);
+void rcc_set_pll_source(uint32_t pllsrc);
+void rcc_set_pllxtpre(uint32_t pllxtpre);
+uint32_t rcc_rtc_clock_enabled_flag(void);
+void rcc_enable_rtc_clock(void);
+void rcc_set_rtc_clock_source(enum rcc_osc clock_source);
+void rcc_set_adcpre(uint32_t adcpre);
+void rcc_set_ppre2(uint32_t ppre1);
+void rcc_set_ppre1(uint32_t ppre1);
+void rcc_set_hpre(uint32_t hpre);
+void rcc_set_usbpre(uint32_t usbpre);
+void rcc_set_prediv(uint32_t prediv);
+uint32_t rcc_system_clock_source(void);
+void rcc_clock_setup_in_hsi_out_64mhz(void);
+void rcc_clock_setup_in_hsi_out_48mhz(void);
+void rcc_clock_setup_in_hsi_out_24mhz(void);
+void rcc_clock_setup_in_hse_8mhz_out_24mhz(void);
+void rcc_clock_setup_in_hse_8mhz_out_72mhz(void);
+void rcc_clock_setup_in_hse_12mhz_out_72mhz(void);
+void rcc_clock_setup_in_hse_16mhz_out_72mhz(void);
```

Comment:
don't copy these. I don't care if f1 uses them, no-one else does, and we're not going to expand their usage.

Reply:
It seems f0 also uses rcc_clock_setup functions.

By the way I still think provide pre-defined clock initialization code with specific osc in and sysclk out is convenient.

Reply:
By the way, I totally agree to drop 12Mhz and 16MHz input functions, because I can not test them now. All my boards with GD32F1x0 that is hackable are 8MHz in.